src/mz/src/secrets.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::{fmt::Display, fs::read_to_string};
+
+use anyhow::{bail, Context, Result};
+use clap::{Parser, Subcommand};
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    configuration::ValidProfile,
+    region::{get_provider_region_environment, CloudProviderRegion},
+};
+
+#[derive(Debug, Subcommand)]
+#[clap(group = clap::ArgGroup::new("input").multiple(false))]
+pub enum SecretCommand {
+    /// Alter a secret
+    Alter {
+        /// The name of the secret to create
+        #[clap(short, long)]
+        name: String,
+
+        #[clap(flatten)]
+        contents: Contents,
+    },
+
+    /// Create a new secret
+    Create {
+        /// The name of the secret to create
+        #[clap(short, long)]
+        name: String,
+
+        #[clap(flatten)]
+        contents: Contents,
+    },
+
+    /// Delete a secret
+    Delete {
+        /// The name of the secret to create
+        #[clap(short, long)]
+        name: String,
+    },
+
+    // List all secrets
+    List,
+}
+
+impl SecretCommand {
+    pub(crate) async fn execute(
+        &self,
+        profile: ValidProfile<'_>,
+        region: CloudProviderRegion,
+        client: Client,
+    ) -> Result<()> {
+        match self {
+            SecretCommand::Alter { name, contents } => {
+                let contents = contents.get()?;
+                let sql = Sql {
+                    query: format!("alter secret {name} as '{contents}';"),
```

Comment:
`name` and `contents` ought to be escaped. You can use [`postgres_protocol::escape::escape_identifier`](https://docs.rs/postgres-protocol/latest/postgres_protocol/escape/fn.escape_identifier.html) to escape the name. `contents` you might be able to pass as a `$1` literal (I forget if allowed); if not use [`postgres_protocol::escape::escape_literal`](https://docs.rs/postgres-protocol/latest/postgres_protocol/escape/fn.escape_literal.html).